src/main/java/com/springboot/star_wars_character_popularity/app/config/SecurityConfiguration.java

```diff
@@ -0,0 +1,64 @@
+package com.springboot.star_wars_character_popularity.app.config;
+
+import com.springboot.star_wars_character_popularity.app.service.UserService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.security.authentication.dao.DaoAuthenticationProvider;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.builders.WebSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.crypto.bcrypt.BCryptPasswordEncoder;
+import org.springframework.security.web.util.matcher.AntPathRequestMatcher;
+
+@Configuration
+@EnableWebSecurity
+public class SecurityConfiguration extends WebSecurityConfigurerAdapter {
+
+   @Autowired
+   private UserService userService;
+
+   @Bean
+   public BCryptPasswordEncoder passwordEncoder(){
+      return new BCryptPasswordEncoder();
+   }
+
+   @Override
+   protected void configure(AuthenticationManagerBuilder auth) throws Exception{
+      auth.authenticationProvider(authenticationProvider());
+   }
+
+   //added support for Spring Data JPA and Hybernate - in Spring Security
+   public DaoAuthenticationProvider authenticationProvider(){
+      DaoAuthenticationProvider auth = new DaoAuthenticationProvider();
+      auth.setUserDetailsService(userService);
+      auth.setPasswordEncoder(passwordEncoder());
+      return auth;
+   }
+
+   @Override
+   public void configure(HttpSecurity http) throws Exception{
+      http.authorizeRequests().
+              antMatchers("/api/user/registration**","/api/user/show-registration-form**","/api/characterlistdb**","/api/characterlistdb/fetch-characters**","/js/**","/css/**","/img/**").permitAll().
```

Comment:
you should divide this into separate antMatchers, so it would be more readable. ( for every path new antMatcher )